### PR TITLE
fix: fix the unexpected behavior of the ConfirmDialog popup

### LINKED ISSUE #4251: Unexpected behavior of the ConfirmDialog component during dragging process
### Describe the bug

If the user adds a margin to the component (`<ConfirmDialog class="mx-4"/>`), the ConfirmDialog component will have unexpected behavior during the dragging process. When the user tries to drag the popup to the left, it moves to the right. The reason for this bug is inside the logic of the calculation coordinates of the popup during the dragging process. We can solve this problem by subtracting the `margin-left` from `leftPos` and `margin-top` from `topPos` when we calculate `this.container.style.left` and `this.container.style.top`. 

This issue linked to the fixes for PrimeNg and PrimeReact

-  https://github.com/primefaces/primeng/issues/13493 - Prime Ng
-  https://github.com/primefaces/primereact/pull/4760 - Prime React

### Reproducer

https://stackblitz.com/edit/kex6xr?file=src%2FApp.vue

### PrimeVue version

3.30.2

### Vue version

3.x

### Language

ES6

### Build / Runtime

Vite

### Browser(s)

Chrome

### Steps to reproduce the behavior

1.. Create a new Vue component and add the `ConfirmDialog `component to the template.
2. Add a button to the template to trigger the popup appearing.
2. Add class `"mx-4"` to the `ConfirmDialog `component.
3. Click the button to popup appears.
4. After the popup appears try to drag it to the left.
5. The popup moves to the right.

[unexpected_behavior.webm](https://github.com/primefaces/primevue/assets/28844263/8d2dfa1d-38b9-4712-9fe1-eec64397ee77)

### Expected behavior

If you drag the popup, it should move in the direction of dragging regardless of the margins of this popup.

[expected_behavior.webm](https://github.com/primefaces/primevue/assets/28844263/e8391836-50af-47a3-a81c-979cd86a9a2a)

### PR DESCRIPTION
**Pull request by Dmytro-Lukashenko**
Fixes #4251  

This issue linked to the fixes for PrimeNg and PrimeReact

-  https://github.com/primefaces/primeng/issues/13493 - Prime Ng
-  https://github.com/primefaces/primereact/pull/4760 - Prime React

### CODE REVIEW

**Comment by Dmytro-Lukashenko** [comment]
Hi @tugcekucukoglu. The main problem is in the Dialog component logic. But other components that extend Dialog have the same problem.

**Comment by Dmytro-Lukashenko** [comment]
Hi @cagataycivici, @tugcekucukoglu, @mertsincan, @bahadirsofuoglu, @yigitfindikli, @Merve7, @onursenture, @atakantepe. Guys, could you please check out this bug? We use the amazing Primevue library in our project and we need to fix it. Thanks. Happy coding!

**Comment by Dmytro-Lukashenko** [comment]
This issue linked to the fixes for Prime,Ng and PrimeReact

primefaces/primeng#13493

primefaces/primereact#4759

File: components/lib/dialog/Dialog.vue [hunk 1]
```diff
@@ -322,18 +322,21 @@ export default {
...
+                            this.container.style.left = (leftPos - marginLeft) + 'px';
                         }
 
                         if (topPos >= this.minY && topPos + height < viewport.height) {
                             this.lastPageY = event.pageY;
-                            this.container.style.top = topPos + 'px';
+                            this.container.style.top = (topPos - marginTop) + 'px';
                         }
                     } else {
                         this.lastPageX = event.pageX;
```
Comment (melloware) [inline components/lib/dialog/Dialog.vue:342]: You missed a spot in the Else clause if `keepInViewPort` is false. See my PrimeReact PR.

**Review by melloware — CHANGES_REQUESTED** [review]

**Review by melloware — APPROVED** [review]

**Comment by Dmytro-Lukashenko** [comment]
Hi @melloware. Could you please merge this PR?

**Comment by mertsincan** [comment]
Thanks a lot for your contribution!

Best Regards,

**Pull request merged**